worker/logforwarder/logforwarder.go

```diff
@@ -125,6 +120,13 @@ func NewLogForwarder(stream LogStream, sender SendCloser) (*LogForwarder, error)
 }
 
 func (lf *LogForwarder) loop() error {
+	if lf.stream == nil {
+		logger.Debugf("log forwarding not enabled")
+		// TODO*ericsnow) restart upon config changed
+		<-lf.catacomb.Dying()
+		return nil
+	}
+
 	records := make(chan logfwd.Record)
 	go func() {
```

Comment:
When we come back to this code, we should think carefully about this. As this goroutine will hang waiting for Next() if we don't have a log message to forward. We really should have some way to interrupt the Next() call, shouldn't we? Or does Next() itself have a tomb to notice that we're shutting down/restarting?